make.js

```diff
@@ -1459,7 +1459,15 @@ target.lint = function() {
   var jshintPath = path.normalize('./node_modules/.bin/jshint');
   if (!test('-f', jshintPath)) {
     echo('jshint is not installed -- installing...');
-    exec('npm install jshint@2.4.x'); // TODO read version from package.json
+    // Read the jshint version to be installed from package.json.
+    fs.readFile('package.json', 'utf8', function (error, data) {
```

Comment:
Can you please read the file synchronously instead, i.e. change `fs.readFile` to `fs.readFileSync`, since the code below relies on `jshint` being immediately available?
Otherwise, when `jshint` isn't installed, linting will fail and require you to do `node make lint` twice (once to install, and once to actually lint).

Obviously, you *could* make the rest of the code async to handle this. But given that `jshint` should only need to be installed *once*, I don't think we should complicate the rest of the function unnecessarily.